core/src/main/java/org/elasticsearch/action/percolate/TransportShardMultiPercolateAction.java

```diff
@@ -160,12 +160,8 @@ public void readFrom(StreamInput in) throws IOException {
             items = new ArrayList<>(size);
             for (int i = 0; i < size; i++) {
                 int slot = in.readVInt();
-                OriginalIndices originalIndices = OriginalIndices.readOriginalIndices(in);
-                PercolateShardRequest shardRequest = new PercolateShardRequest(new ShardId(index, shardId), originalIndices);
-                shardRequest.documentType(in.readString());
-                shardRequest.source(in.readBytesReference());
-                shardRequest.docSource(in.readBytesReference());
-                shardRequest.onlyCount(in.readBoolean());
+                PercolateShardRequest shardRequest = new PercolateShardRequest();
```

Comment:
I tried to remove it but unit tests are using it (PercolateDocumentParserTests), the percolator document parser relies on the shard id to be set on the percolate shard request. So lets keep it?